addons/website_slides/models/slide_slide_resource.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class SlideResource(models.Model):
+    _name = 'slide.slide.resource'
+    _description = "Additional resource for a particular slide"
+
+    slide_id = fields.Many2one('slide.slide', required=True, ondelete='cascade')
+    type = fields.Selection([('file', 'File'), ('url', 'Link')], required=True, default="url")
+    name = fields.Char('Name', required=True)
+    data = fields.Binary('Resource', compute='_reset_resources', store=True)
+    link = fields.Char('Link', compute='_reset_resources', store=True)
+
```

Comment:
Would be great to add a `_sql_constraints`
- if type == 'file', then link should be null
- if type == 'url', then data should be null